includes/data-events/connectors/ga4/class-ga4.php

```diff
@@ -0,0 +1,218 @@
+	 *
+	 * @throws \Exception If the event is invalid.
+	 * @return void
+	 */
+	public static function handle_reader_logged_in( $timestamp, $data, $user_id ) {
+
+		$params    = $data['ga_params'];
+		$client_id = $data['ga_client_id'];
+
+		if ( empty( $client_id ) ) {
+			throw new \Exception( 'Missing client ID' );
+		}
+
+		$event = new Event( 'reader_login', $params );
+		self::send_event( $event, $client_id, $timestamp, $user_id );
+	}
+
+	/**
+	 * Gets the credentials for the GA4 API.
+	 *
+	 * @return array
+	 */
+	private static function get_credentials() {
+		$api_secret     = get_option( 'ga4_api_secret' );
+		$measurement_id = get_option( 'ga4_measurement_id' );
+		return compact( 'api_secret', 'measurement_id' );
+	}
+
+	/**
+	 * Gets the API URL for GA4
+	 *
+	 * @return WP_Error|string
+	 */
+	public static function get_api_url() {
+		$credentials    = self::get_credentials();
+		$api_secret     = $credentials['api_secret'];
+		$measurement_id = $credentials['measurement_id'];
+		if ( ! $api_secret || ! $measurement_id ) {
+			return new WP_Error( 'missing_credentials', 'Missing GA4 API credentials.' );
+		}
+		return add_query_arg(
+			[
+				'api_secret'     => $api_secret,
+				'measurement_id' => $measurement_id,
+			],
+			'https://www.google-analytics.com/mp/collect'
+		);
+	}
+
+	/**
+	 * Sends an event to GA4.
+	 *
+	 * @param Event  $event The event object.
+	 * @param string $client_id The GA client ID obtained from the cookie.
+	 * @param int    $timestamp The timestamp of the event.
+	 * @param string $user_id User identifier. Use Reader_Activation::get_client_id().
+	 * @throws \Exception If the credentials are missing.
+	 * @return void
+	 */
+	private static function send_event( Event $event, $client_id, $timestamp, $user_id = null ) {
+
+		$url = self::get_api_url();
+
+		if ( is_wp_error( $url ) ) {
+			throw new \Exception( $url->get_error_message() );
+		}
+
+		$timestamp_micros = (int) $timestamp * 1000000;
+
+		$payload = [
+			'client_id'        => $client_id,
+			'timestamp_micros' => $timestamp_micros,
+			'events'           => [
+				$event->to_array(),
+			],
+		];
+
+		if ( $user_id ) {
+			$payload['user_id'] = $user_id;
+		}
+
+		wp_remote_post(
+			$url,
+			[
+				'body' => wp_json_encode( $payload ),
+			]
+		);
+
+		self::log( sprintf( 'Event sent - %s - Client ID: %s', $event->get_name(), $client_id ) );
+
+	}
+
+	/**
+	 * Extracts the Client ID from the _ga cookie
+	 *
+	 * @return string
+	 */
+	private static function extract_cid_from_cookies() {
+
+		if ( isset( $_COOKIE['_ga'] ) ) {
+			$cookie_pieces = explode( '.', $_COOKIE['_ga'], 3 ); // phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.cache_constraints___COOKIE, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			if ( 1 === count( $cookie_pieces ) ) {
+				$cid = reset( $cookie_pieces );
+			} else {
+				list( $version, $domain_depth, $cid ) = $cookie_pieces;
+			}
+		} else {
+			$cid = 555;
```

Comment:
```suggestion
			$cid = '555';
```

If we are to keep the "anonymous client", the `cid` should be a `string` or GA's API is unable to parse the JSON payload.

Reply:
let's throw

Reply:
done